digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/process/PedestrianCrossingValidator.scala

```diff
@@ -0,0 +1,106 @@
+package fi.liikennevirasto.digiroad2.process
+
+import fi.liikennevirasto.digiroad2.asset.SideCode.{AgainstDigitizing, TowardsDigitizing}
+import fi.liikennevirasto.digiroad2.{GeometryUtils, Point}
+import fi.liikennevirasto.digiroad2.asset.{AssetTypeInfo, PedestrianCrossings, Private, SideCode}
+import fi.liikennevirasto.digiroad2.dao.pointasset.{OraclePedestrianCrossingDao, PedestrianCrossing, PersistedTrafficSign}
+import fi.liikennevirasto.digiroad2.linearasset.RoadLink
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.service.pointasset.TrafficSignType
+
+class PedestrianCrossingValidator extends AssetServiceValidatorOperations {
+  override type AssetType = PedestrianCrossing
+  override def assetTypeInfo: AssetTypeInfo =  PedestrianCrossings
+  override val radiusDistance: Int = 50
+
+  lazy val dao: OraclePedestrianCrossingDao = new OraclePedestrianCrossingDao()
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  val allowedTrafficSign: Set[TrafficSignType] = Set(TrafficSignType.PedestrianCrossing)
+
+  override def filteredAsset(roadLink: RoadLink, assets: Seq[AssetType], pointOfInterest: Point, distance: Double, trafficSign: Option[PersistedTrafficSign] = None): Seq[AssetType] = {
+    def assetDistance(assets: Seq[AssetType]): Option[(AssetType, Double)] = {
+      val (first, _) = GeometryUtils.geometryEndpoints(roadLink.geometry)
+      val isAdjacente = GeometryUtils.areAdjacent(pointOfInterest, first)
+        trafficSign match {
+        case Some(traffic) =>
+          val filteredAsset = if(SideCode.apply(traffic.validityDirection) == TowardsDigitizing)
+            assets.filter(a => a.linkId == roadLink.linkId && a.mValue >= traffic.mValue)
```

Comment:
```
val filteredAsset = if(SideCode.apply(traffic.validityDirection) == TowardsDigitizing)
            assets.filter(a => a.linkId == roadLink.linkId && a.mValue >= traffic.mValue)
          else
            assets.filter(a => a.linkId == roadLink.linkId && a.mValue <= traffic.mValue)
```

The filters are both filtering by the equal `=` should it be in that way? Can't we use the partition instead of using 2 filters?